plugins/languages/vimtex.nix

```diff
@@ -24,21 +24,56 @@ in
         '';
         default = {};
       };
+
+      viewMethod = mkOption {
+        type = types.str;
+        description = ''
+          The view method that vimtex will use to display PDF's.
+          Check https://github.com/lervag/vimtex/blob/03c83443108a6984bf90100f6d00ec270b84a339/doc/vimtex.txt#L3322
+          for more information.
+        '';
+        default = "general";
+      };
+      installTexLive = mkOption {
+        type = types.bool;
+        description = ''
+                 Whether or not to install TexLive.
+          See https://nixos.wiki/wiki/TexLive.
+        '';
+        default = false;
+      };
+      texLivePackage = helpers.mkPackageOption "texLivePackage" pkgs.texlive.combined.scheme-medium;
     };
 
     config = let
       globals =
         {
           enabled = cfg.enable;
           callback_progpath = "nvim";
+          view_method = cfg.viewMethod;
         }
         // cfg.extraConfig;
+      basePackages =
+        [pkgs.pstree]
+        ++ (
+          if cfg.installTexLive
+          then [cfg.texLivePackage]
+          else []
+        );
```

Comment:
you can do `lib.optional cfg.installTexLive cfg.texLivePackage`

Reply:
Very nice, I was wondering if there was a better way. Fixed